### ISSUE TITLE
`extends` API of tslint not supported in NodeJs API

### ISSUE THREAD

**Issue by kratiahuja** [palantir/tslint#2011]
### Bug Report

- __TSLint version__: `4.3.1`
- __TypeScript version__: `2.1.4`
- __Running TSLint via__: Node.js API 

#### TypeScript code being linted

```ts
var Xx = "abcd"; 
```

with `tslint.json` configuration: 

```json
{
    extends: "tslint:recommended"
}
```

#### NodeJs Script example:
```javascript
var Linter = require('tslint').Linter;
var fs = require('fs');

var fileName = "tests/fixtures/errorFiles/errorFile1.ts";
var configuration = {
  /*rules: {
      "no-trailing-whitespace": true
  }*/
  "extends": "tslint:recommended"
};
var options = {
    formatter: "json",
    rulesDirectory: "customRules/",
    formattersDirectory: "customFormatters/"
};

var fileContents = fs.readFileSync(fileName, "utf8");
var linter = new Linter(options);
linter.lint(fileName, fileContents, configuration);
var result = linter.getResult();

console.log(result);
```

#### Actual behavior

When running the above script with the tslint configuration only set to using `extends`, it throws an error: `No valid rules have been specified`.  It looks like the `extends` configuration is only working in tslint-cli since it _only_ invokes `findConfiguration` method in CLI and not the NodeJS API. 

#### Expected behavior

If using the `lint` public API of tslint (via NodeJs) and having the configuration file to only have `extends`, it should lint the file correctly with the extended rules.

**Comment by kratiahuja** [comment]
I looked at this. The way the `lint` API is designed, there is no way for to call `findConfiguration` since that needs a configuration path. I ended up fixing my broccoli plugin by explicitly calling `findConfiguration`.

My suggestion is to update the README doc with the same as well. I am going to send a PR to update the README example with the same.
